src/array.cr

```diff
@@ -1535,57 +1535,106 @@ class Array(T)
     self
   end
 
-  # Returns an array with all elements in the collection sorted.
+  # Returns a new array with all elements sorted.
   #
   # ```
   # a = [3, 1, 2]
   # a.sort # => [1, 2, 3]
   # a      # => [3, 1, 2]
   # ```
-  #
-  # Optionally, a block may be given that must implement a comparison, either with the comparison operator `<=>`
-  # or a comparison between *a* and *b*, where a < b yields -1, a == b yields 0, and a > b yields 1.
-  def sort
+  def sort : Array(T)
     dup.sort!
   end
 
-  def sort(&block : T, T -> Int32)
+  # Returns a new array with all elements sorted based on the return value of
+  # the block given.
```

Comment:
Maybe "the given block used as comparator"? Makes the difference to sort_by more clear.

Reply:
`...block used as comparator..` does not read quite right to me. It is the return value (or what the block yields) that is used as the comparator.

> Returns a new array with all elements sorted. The given block is called for each element, then the comparison method `#<=>` is called on the object returned from the block to determine sort order.

Is that too much of a mouthful?

Reply:
This would describe `sort_by`. Here the block is actually used as comparator instead of `<=>`.

Reply:
ahh, my bad, I saw "sort_by" in your original comment, and ran with that.